arch/common/sw_isr_common.c

```diff
@@ -16,8 +16,6 @@ void z_isr_install(unsigned int irq, void (*routine)(void *), void *param)
 {
 	unsigned int table_idx = irq - CONFIG_GEN_IRQ_START_VECTOR;
 
-	__ASSERT(!irq_is_enabled(irq), "IRQ %d is enabled", irq);
```

Comment:
Can we take this commit that removes the assert out of this PR? Seems we have not reached an agreement about that, maybe do it on  a separate commit

Reply:
I suppose you mean a separate PR; the problem here is that the test will not pass without that commit.

Reply:
OK, I thought that was only a problem in Cortex-R

Reply:
This is really a problem with GIC (and all the platforms that use GIC; i.e. Cortex-A and Cortex-R), as described here:
https://github.com/zephyrproject-rtos/zephyr/pull/23518#issuecomment-601072336

Reply:
So I want either to reach the agreement among all ARCH maintainers to remove this assert, or remove it conditionally for GIC, or even ARM && !CORTEX_M

Reply:
I still have doubts about the effectiveness of this assertion as [I already explained earlier](https://github.com/zephyrproject-rtos/zephyr/pull/23518#issuecomment-601077127), but I will make it `#ifndef CONFIG_GIC` for now so that we are not blocked by it.

Reply:
Fixed

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/23518#issuecomment-606632953